### PR TITLE
feat(plans): testing harness + renderer refactor + observability (v0.107.0)

### PR DESCRIPTION
**Pull request by yuens1002**
## Summary

Session 1 of the provider-plan-sdk-alignment rebuild. Splits **renderer correctness** (does the component project a typed input correctly?) from **resolver fidelity** (does the provider actually return what we think it returns?). Ships the testing layers and the minimum runtime observability that makes drift visible after deploy.

**Three architectural decisions** crystallized in this session and now baked into the code + docs:

1. **Test the renderer's contract, not the data.** Component tests construct typed inputs inline; fixtures are examples, not specs. Hand-derived scenario compositions exist for dev iteration + screenshots only. See `architecture.md` D4.
2. **Role separation — SDK = shape, Provider = decisions, Store = placement.** Encoded as a roles table; the store never derives state, computes exhaustion, infers cadence, or validates business config. See `architecture.md` §2.
3. **Compatibility-over-time is signal-triggered, not timeline-triggered.** Self-hosted is live but unstressed (zero current customers); hosted/trial gate isn't open. Minimum observability (`[plans.fetch.failed]` / `[plans.empty.unexpected]` logs) is the runtime safety net; the heavier machinery (Zod boundary validation, version handshake, admin banner, fallback card, deprecation policy) ships when a real signal appears. See `architecture.md` D13 + §9.

## Code

- **\`formatters.ts\`** — pure transforms (price, date, pool count, days remaining); 21 unit tests.
- **\`_components/PoolCtaMenu.tsx\`** — shared 3-dot dropdown used by ActiveCard / TrialCard / ExpiredCard. Respects \`action.disabled\` + \`action.disabledReason\` tooltip. Closes the gap where Trial/Expired cards previously ignored \`pool.cta\` entirely.
- **\`PlanPageClient.tsx\`** refactored to consume both.
- **\`page.tsx\`** — \`?scenario=<key>\` dev override gated on \`NODE_ENV\`.
- **\`lib/plans.ts\`** — structured \`[plans.fetch.failed]\` / \`[plans.empty.unexpected]\` boundary logs; ~30 lines, no new deps.

## Tests

- **96 component contract tests** across 6 per-state files (NoneCard, ActiveCard, TrialCard, ExpiredCard, CancelledCard, InactiveCard) — hermetic, runs in <2s.
- **21 formatter unit tests** covering happy path + edge cases (days-remaining singular/plural, expired sale, undefined \`purchased\`).
- **SDK SCAFFOLD pin** (\`sdk-scaffold-pins.test.ts\`) — snapshot pin with date normalisation.
- **\`fixtures/plan-scenarios.ts\`** — SDK-derived dev fixtures for the \`?scenario=\` override and screenshot harness.
- **Total**: 1388 tests pass (up from 1316). Typecheck clean. Precheck clean (1 pre-existing TanStack warning, untouched).

## Scripts

- **\`scripts/capture-plan-scenarios.ts\`** + \`npm run plans:capture\` — captures prod payloads to \`e2e/plans/captured/\` for the future Layer 4 captured-payload Jest test (filed as ST-4 in the deferred tracker).
- **\`scripts/screenshot-plan-scenarios.ts\`** rewritten as the single canonical visual harness — replaces 8 throwaway iteration scripts (deleted).

## Docs

Three new docs under \`docs/features/provider-plan-sdk-alignment/\`:

- **\`architecture.md\`** — store-side view of the cross-repo contract; roles table; 4 drift surfaces; 6 testing layers; per-field data ownership; §9 today's state + minimum observability; §10 provider-side concerns out of scope; decisions log D1–D13.
- **\`plan.md\`** — Session 1 deliverables; Session 2 (CONVERTING + ConversionModal) cross-repo deliverables table; comprehensive deferred-work tracker (9 store-side, 7 provider-side, 6 SDK-side RFCs including SDK-RFC-MCP-VER bundled with next SDK touch).
- **\`session-1/ACs.md\`** — 17 acceptance criteria with presence/absence/placement format on every How column. Ship-gating subset: 10 ACs PASS.

Original session 1–5 work archived under \`docs/features/provider-plan-sdk-alignment/archive/\`.

## Customer impact

The hosted/trial gate is **not** open to customers today, so previously-flagged hosted/trial UX gaps don't reach paying users. Self-hosted is live in production with zero current customers — Session 1's code paths are correct against verified contract tests, and any future signup hits a tested renderer with observability hooks in place.

## Test plan

- [x] \`npm run typecheck\` — 0 errors
- [x] \`npm run test:ci\` — 122 suites, 1388 tests, 0 failures
- [x] \`npm run precheck\` — 0 errors (1 pre-existing warning)
- [x] Component contract tests pass — 96 tests across 6 per-state files
- [x] Formatter unit tests pass — 21 tests
- [x] SDK SCAFFOLD pin holds
- [x] Manual dev verification of \`?scenario=<key>\` override on \`/admin/support/plans\`
- [ ] Copilot review + comments addressed + threads resolved
- [ ] Squash-merge to main
- [ ] Phase B: tag \`v0.107.0\` + GitHub Release (triggers in-app upgrade banner)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -87,6 +87,7 @@
     "@vercel/blob": "^2.3.0",
     "@vercel/speed-insights": "^1.3.1",
     "archiver": "^7.0.1",
+    "artisan-roast-sdk": "file:../artisan-roast-sdk",
     "bcryptjs": "^3.0.3",
     "class-variance-authority": "^0.7.1",
```
[the remark this answers is not in the document]
Comment (yuens1002) [reply package.json:92]: (reply)

[the remark this answers is not in the document]
Comment (yuens1002) [reply package.json:92]: test

[the remark this answers is not in the document]
Comment (yuens1002) [reply package.json:92]: Acknowledged — this is a longstanding repo convention (introduced in `61ce781a feat(sdk-align): install artisan-roast-sdk file dep`, shipped through v0.105.x and v0.106.x). The SDK is intentionally not yet published to a registry; consumers (including Vercel builds) work because the SDK is colocated during build.

Filed for follow-up: the deferred-work tracker in [`plan.md`](https://github.com/yuens1002/artisan-roast/blob/feat/plan-scenario-corrections/docs/features/provider-plan-sdk-alignment/plan.md) names this as part of compatibility-over-time. Bumping the SDK to a published package is the right move when self-hosted launches; today it's deferred under D13 (signal-triggered, not timeline-triggered).

Not changing in this PR — orthogonal to the testing-harness work, will revisit in the SDK packaging session.
